src/Proxy/DriveItemProxy.php

```diff
@@ -295,6 +296,54 @@ public function upload($name, $content, array $options = [])
 
         return new self($this->graph, $driveItem);
     }
+	
+	/**
+     * Creates a upload session to upload a file under this folder drive item.
+     *
+     * @param string $name
+     *        The name.
+     * @param string|resource|\GuzzleHttp\Psr7\Stream $content
+     *        The content.
+     * @param array $options
+     *        The options.
+     *
+     * @return UploadSessionProxy
+     *         The upload session created.
+     *
+     * @todo Support content type in options.
+     */
+    public function startUpload($name, $content, array $options = [])
+    {
+		$name         = rawurlencode($name);
+        $driveLocator = "/drives/{$this->parentReference->driveId}";
+        $itemLocator  = "/items/{$this->id}";
+		$endpoint = "$driveLocator$itemLocator:/$name:/createUploadSession";
+		
+		//Include extended options with existing item for upload session. Extended options to be overwritten if not mentioned
+		$item = [
+			'name' => isset($options['name']) ? $options['name'] : $name,
+			'description' => isset($options['name']) ? $options['description'] : '',
+			'@microsoft.graph.conflictBehavior' => isset($options['conflictBehavior']) ? $options['conflictBehavior'] : 'rename',
+		];
```

Comment:
When I suggested using the `+` operator to merge options, I meant to do something like this: https://github.com/krizalys/onedrive-php-sdk/blob/master/src/Proxy/DriveItemProxy.php#L183

The idea is to have an array with some defaults (constants, or coming from function parameters - `$body` in the sample code) and "add" to them the `$options` given, so the latter options take precedence if set.

Example based on your code:

```php
$body = [
    'name' => (string) $name,
    'item' => [
        '@microsoft.graph.conflictBehavior' => 'rename',
        // more default options here...
    ],
    // or here...
];

// and latter, when calling attachBody:
    ->attachBody($body + $options)
```

BTW, if `description` is optional for the OneDrive API, you do not need to set it to an empty string and should remove it.

Reply:
Problem with options array being common across uploadSession POST request is that, this request accept only certain parameters passed in body. If anything apart from that is passed, it throws an error.

That's why I can't use merge array, instead I take the necessary option available in options for POST request.
```php
$item = [
    'name' => $name,
    'description' => isset($options['name']) ? $options['description'] : '',
    '@microsoft.graph.conflictBehavior' => isset($options['conflictBehavior']) ? $options['conflictBehavior'] : 'rename',
];
```
The same options is used later in UploadSessionProxy for getting streamSize and chunkSize from the array.

Can I refactor the above code like this,
```php
$item = [
    'name' => $name,
    '@microsoft.graph.conflictBehavior' => isset($options['conflictBehavior']) ? $options['conflictBehavior'] : 'rename',
];
if(isset($options['description'])) {
    $item = $item + ['description' => $options['description']];
}
```

Reply:
@krizalys Can you check on updated code